### ISSUE TITLE
Facebook posts always have an image

### ISSUE THREAD

**Issue by fritzmg** [terhuerne/Contao-Aggregator#11]
The following code in [AggregatorEngine](https://github.com/terhuerne/Contao-Aggregator/blob/1.0.3/classes/AggregatorEngine.php#L438) (lines 438 through 449)
```php
$cacheLibrary[$count]['item']['picture'] = str_replace('/s130x130', '', str_replace('_s', '_n', $item['picture']));
if(strpos($cacheLibrary[$count]['item']['picture'], 'safe_image.php') !== false) {
	$parts = parse_url($cacheLibrary[$count]['item']['picture']);
	parse_str($parts['query'], $query);
	$cacheLibrary[$count]['item']['picture'] = $query['url'];
}else{
	$ch = curl_init('https://graph.facebook.com/v2.2/'.$item['object_id'].'/picture?type=normal&access_token='.$GLOBALS['TL_CONFIG']['aggregator_facebook_app_id'].'|'.$GLOBALS['TL_CONFIG']['aggregator_faceboook_app_secret']);
	curl_exec($ch);
	$response = curl_getinfo($ch);
	$cacheLibrary[$count]['item']['picture'] = $response['redirect_url'];
	curl_close($ch);
}
```
causes Facebook posts to _always_ have an image, even if the actual Facebook post does not. In such a case, [this image](https://fbstatic-a.akamaihd.net/rsrc.php/v2/yA/r/gPCjrIGykBe.gif) will be shown, since that is the image, that Facebook returns in this API call
```php
$ch = curl_init('https://graph.facebook.com/v2.2/'.$item['object_id'].'/picture?type=normal&access_token='.$GLOBALS['TL_CONFIG']['aggregator_facebook_app_id'].'|'.$GLOBALS['TL_CONFIG']['aggregator_faceboook_app_secret']);
```
whenever no actual picture is present.

**Comment by terhuerne** [comment]
Good point. I'm going to include a fix in the next release.

**Comment by kossuth** [comment]
//ALL FIELDS  Example ::id,admin_creator,application,call_to_action,caption,created_time,description,feed_targeting,from,icon,is_hidden,is_published,link,message,message_tags,name,object_id,picture,place,privacy,properties,shares,source,status_type,story,story_tags,targeting,to,type,updated_time,with_tags

//variable control ;)							$fields="id,created_time,description,attachments,link,message,picture,name,object_id,type";

Edit code Aggegator in checkForUpdates() function
$data = $this->fetchUrl('https://graph-video.facebook.com/v2.4/'.urlencode($allActiveJobs->facebookUser).'/posts/?attachments&fields='.$fields.'&access_token='.$GLOBALS['TL_CONFIG']['aggregator_facebook_app_id'].'|'.$GLOBALS['TL_CONFIG']['aggregator_faceboook_app_secret']);

example structure :: return .::::
"id"::'#####
"created_time": ###
 "name": #####,
         "object_id": #####,
         "type": "photo",
         "attachments": {
            "data": [
               {
                  "subattachments": {
                     "data": [
                        {
                           "description": "Der richtige Fokus ist entscheidend f\u00fcr einen Entomologen...",
                           "media": {
                              "image": {
                                 "height": 720,
                                 "src": ### External Link ##########
                                 "width": 542
                              }
                           },			

edit code : Picture  in parseDataToCache() function 
Easy reference first element for Array.

$linkImgFBExt="";
if(isset($item['attachments']['data'][0]['subattachments']['data'][0]['media']['image']['src']))
$linkImgFBExt=$item['attachments']['data'][0]['subattachments']['data'][0]['media']['image']['src'];
else
$linkImgFBExt=$item['attachments']['data'][0]['media']['image']['src'];


$cacheLibrary[$count]['item']['picture'] = $linkImgFBExt;

References:
https://developers.facebook.com/docs/graph-api/reference/v2.4/post
https://developers.facebook.com/docs/graph-api/reference/v2.4/story_attachment

**Comment by terhuerne** [comment]
Facebook has a new picture API, which makes my hook into getting the original image obsolete. This will be included in an update, which will be released this afternoon.

**Comment by terhuerne** [comment]
Fixed with Version 1.1. Should be available in the next minutes via composer.
